drivers/spi/spi_cc23x0.c

```diff
@@ -63,6 +66,22 @@ struct spi_cc23x0_data {
 #endif
 };
 
+static inline void spi_cc23x0_pm_policy_state_lock_get(void)
+{
+#ifdef CONFIG_PM_DEVICE
+	pm_policy_state_lock_get(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
+	pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
```

Comment:
is there a reason you are not using device constraints from DT instead of hardcoding these modes which might be different per SOC